src/main/scala/nl.knaw.dans.easy.bagstore/BagFacadeComponent.scala

```diff
@@ -146,9 +146,9 @@ trait Bagit4FacadeComponent extends BagFacadeComponent {
 
     override def getPayloadFilePaths(bagDir: Path): Try[Set[Path]] =  {
       getBag(bagDir)
-        .map(_.getPayloadManifests.asScala.headOption.map(_.asScala.map {
+        .map(_.getPayloadManifests.asScala.map(_.asScala.map {
           case (path, _) => Paths.get(path)
-        }.toSet).getOrElse(Set.empty))
+        }.toSet)).map(_.reduce(_ ++ _))
```

Comment:
with the `Buffer[Set[...]].reduce` you assume that the buffer is NOT empty. If you're not sure about that, change the `reduce` to `foldLeft(Set.empty[...])(_ ++ _)`